lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -159,34 +169,22 @@ void Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
     Tensor *w = getTensorByName(op.input(1));
 
     // Transpose the weights to the right format. Glow expects to read the
-    // weights in the format CRSK. Caffe2 stores the operators as KCRS.
+    // weights in the format CRSK.
     // C - output_depth, R - filter_height, S - filter_width, K - input_depth.
+    // Caffe2 "Conv" op always stores the weight as KCRS, while for "Int8Conv",
+    // and "Int8ConvRelu", the weights always follows the "order" arg.
     Tensor wtag;
-    w->transpose(&wtag, NCHW2NHWC);
+    if (typeName != "Conv" && order == "NHWC") {
```

Comment:
Only NHWC order is supported by Int8Conv:
https://github.com/pytorch/pytorch/blob/9cb4bce847af688ca344caaa876d3c47a36f28bf/caffe2/operators/quantized/int8_conv_op.h#L25

Also, are you sure that "order" arg applies to weights too, not just input?
Reading the docs, it seems to me that weight always has to be transposed, because it's outC x inC x h x w
https://github.com/pytorch/pytorch/blob/9cb4bce847af688ca344caaa876d3c47a36f28bf/caffe2/operators/quantized/int8_conv_op.cc#L49